### ISSUE TITLE
AbelianPcpGroup with list containing 1 does not behave properly

### ISSUE THREAD

**Issue by stertooy** [gap-packages/polycyclic#42]
Commit 0ff3b3556e6871757eb575328b187f346f8c8b5e introduces the following error:
```
gap> TorsionSubgroup( AbelianPcpGroup( [0,1,2] ) );
Error, wrong reduction in ExponentsByPcp at /usr/lib/gap/pkg/polycyclic-master/gap/basic/pcpexpo.gi:184 called from
ExponentsByPcp( pcp, pcp[i] ^ rels[i] ) at /usr/lib/gap/pkg/polycyclic-master/gap/basic/pcppcps.gi:486 called from
CyclicDecomposition( pcp ) at /usr/lib/gap/pkg/polycyclic-master/gap/basic/pcppcps.gi:638 called from
Pcp( G, "snf" ) at /usr/lib/gap/pkg/polycyclic-master/gap/pcpgrp/torsion.gi:23 called from
TorsionSubgroupAbelianPcpGroup( G ) at /usr/lib/gap/pkg/polycyclic-master/gap/pcpgrp/torsion.gi:101 called from
<function "TorsionSubgroup for pcp groups">( <arguments> )
 called from read-eval loop at *stdin*:1
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
```
while for older versions this should give
```
gap> TorsionSubgroup( AbelianPcpGroup( [0,1,2] ) );
Pcp-group with orders [ 2 ]
```
The problem seems to lie with the "1", I haven't encountered any problems with lists not containing ones.

**Comment by fingolfin** [comment]
The "1" wound indicate a trivial factor. We could just remove any "1"s from the list, or alternatively, print an error if one is encountered.

**Comment by sTertooy** [comment]
IMHO removing the ones is the better solution. On the one hand, I don't think an error should be thrown for what is essentially correct input; on the other hand, `AbelianPcpGroup` already "fixes" bad input in other cases (negative integers given as input are replaced with `0`), so this feels more consistent with that behaviour.

**Comment by fingolfin** [comment]
Another related bug, already present in older versions:
```
gap> A:=AbelianPcpGroup( [0,1,2] );
Pcp-group with orders [ 0, 1, 2 ]
gap> Order(A.2);  # <-- does not terminate
```

Moreover, when trying to figure out what semantics we promise, I realized that the documentation for `AbelianPcpGroup` is wrong or at least incomplete (it only describe a two argument version which I wasn't even aware of ;-). I'll fix that ASAP.

Anyway, having thought a bit more about it, I am very reluctant to just drop that generator for 1, as I am afraid it will cause difficult to find bugs in code using polycyclic: if code asks for an abelian group with a specific number of generators, then it seems unwise to change that. So I really think we either need to fix support for order 1 generators, or else produce an error.

I note that e.g. pc groups seems to deal fine with generators of order 1; `AbelianGroup(IsPcGroup, [1,2]);` produce a result that seems to work well. That code does something "clever" (?): internally, there is only one generator, `f1`; but `GeneratorsOfGroup` returns two generators:  `id` and `f1`. Doing something similar for pcp groups would require modifying `PcpGroupByCollectorNC` resp. adding a special variant. Certainly doable; but I am not yet convinced this is a good idea.

Hence I think I'll change this into an error after all. Because I don't like making a guess about what one should do in a weird edge case; because once we return a certain thing in this case, people may start depending it, and we have no chance to revise it. But if we produce an error, we can always change that from an error to something that actually works -- this can't break existing code. So, I'd rather wait for people to complain, because then we can talk to them to figure out if they have a legit usecase, and then implement a solution that actually supports that use case (or else suggest to them an alternative solution).

**Comment by stertooy** [comment]
This seems to have been fixed thanks to #79 and #81, so I'll close this.
